dotnet/private/rules/csharp/actions/csharp_assembly.bzl

```diff
@@ -51,6 +51,24 @@ def _write_internals_visible_to_csharp(actions, label_name, dll_name, others):
 
     return output
 
+def _collect_analyzer_dependencies(deps):
+    """Collect the runtime libraries of this analyzer. These will be passed to the compiler.
+
+    Args:
+        deps: The list of dependencies of the analyzer.
+
+    Returns:
+        The list of analyzer dependencies.
+    """
+
+    dlls = []
+    for dep in deps:
+        runtime_info = dep[DotnetAssemblyRuntimeInfo]
+
+        # FIXME: Should this respect `not strict_deps`?
```

Comment:
What does MSBuild do here by default? Does it add the transitive deps as well?

Reply:
No, it requires the transitive dependencies too, using `TargetPathWithTargetPlatformMoniker`.

Reply:
Ok, then let's keep it the same here. It can be confusing though so maybe we end up creating a separate rule/macro for analyzer creation since the rules are not really the same as for normal dlls/binaires.